airflow/providers/ftp/hooks/ftp.py

```diff
@@ -248,7 +248,7 @@ def get_mod_time(self, path: str) -> datetime.datetime:
         except ValueError:
             return datetime.datetime.strptime(time_val, '%Y%m%d%H%M%S')
 
-    def get_size(self, path):
+    def get_size(self, path: str) -> Union[int, str]:
```

Comment:
Just returns int as it is the size

```suggestion
    def get_size(self, path: str) -> int:
```

Reply:
because of implicit return of string from size() I have put an option of either int or str.

Reply:
https://docs.python.org/3/library/ftplib.html#ftplib.FTP.size

It will size in bytes or None, so 

```suggestion
    def get_size(self, path: str) -> Optional[int]:
```

It won't return string

Reply:
True, but mypy throws an error for int return type as there is an implicit string print statement in this function. I can change it to `Any` (but that will again a dynamic typed one)

Reply:
Mypy is going crazy, I would suggest keeping it as 

```
def get_size(self, path: str) -> Optional[int]:
```

and do a cast on `return conn.size(path)` as below on L259:

```
return int(conn.size(path))
```

Reply:
Your suggestion needs explicit casting of return statement to `int` 
```suggestion
    def get_size(self, path: str) -> Optional[int]:
        conn = self.get_conn()
        return int(conn.size(path) or 0)
```

Reply:
Well, I would suggest to not do type checking for this function as we have more than 75% coverage for FTP provider. What's your thought?

Reply:
or 
```suggestion
    def get_size(self, path: str) -> Optional[int]:
        conn = self.get_conn()
        size = conn.size(path)
        return int(size) if size else None
```

Reply:
that works 👍🏼 Will make the changes!

Reply:
The above suggestion should work, just tested it with mypy. 
```
Run mypy.................................................................Passed
```

Once that is done, let's merge this one :)

Reply:
Thanks brother, let me start my journey into Airflow from here :)